frontend/src/Frontend/UI/Dialogs/Send.hs

```diff
@@ -203,15 +280,47 @@ sendConfig model sender = Workflow $ do
           Left e -> do
             divClass "group" $ text $ e <> ": please go back and check the recipient."
             pure $ pure Nothing
-          Right _ -> do
-            elClass "h2" "heading heading_type_h2" $ text "Gas Payer"
-            divClass "group" $ elClass "div" "segment segment_type_tertiary labeled-input" $ do
+          Right (ka, _amount) -> do
+            let toChain = _kadenaAddress_chainId ka
+                fromChain = _account_chainId fromAccount
+            when (toChain /= fromChain) $ do
+              elClass "h3" ("heading heading_type_h3") $ text "This is a cross chain transfer."
+              el "p" $ text $ T.concat
+                [ "Coin will be transferred from chain "
+                , _chainId fromChain
+                , " to chain "
+                , _chainId toChain
+                , "."
+                ]
+              el "p" $ text $ T.unwords
+                [ "This is a multi step operation."
+                , "The coin will leave the sender account immediately, and gas must be paid on the recipient chain in order to redeem the coin."
+                ]
+            elClass "h2" "heading heading_type_h2" $ do
+              text "Gas Payer"
+              when (toChain /= fromChain) $ text $ " (Chain " <> _chainId fromChain <> ")"
+            fromGasPayer <- divClass "group" $ elClass "div" "segment segment_type_tertiary labeled-input" $ do
               divClass "label labeled-input__label" $ text "Account Name"
               let cfg = def & dropdownConfig_attributes .~ pure ("class" =: "labeled-input__input")
-                  chain = pure $ Just $ _account_chainId sender
+                  chain = pure $ Just fromChain
               gasAcc <- uiSenderDropdown cfg never model chain
               let toAccount ma m = find ((ma ==) . Just . _account_name) $ fmapMaybe (\case SomeAccount_Account a -> Just a; _ -> Nothing) m
               pure $ toAccount <$> gasAcc <*> model ^. wallet_accounts
+            toGasPayer <- if toChain == fromChain
+              then pure $ pure $ Just Nothing
+              else do
+                -- TODO this bit should have an option with a generic input for Ed25519 keys
+                -- and perhaps be skippable entirely in favour of a blob the user
+                -- can send to someone else to continue the tx on the recipient chain
+                elClass "h2" "heading heading_type_h2" $ text $ "Gas Payer (Chain " <> _chainId toChain <> ")"
+                divClass "group" $ elClass "div" "segment segment_type_tertiary labeled-input" $ do
+                  divClass "label labeled-input__label" $ text "Account Name"
+                  let cfg = def & dropdownConfig_attributes .~ pure ("class" =: "labeled-input__input")
+                      chain = pure $ Just toChain
+                  gasAcc <- uiSenderDropdown cfg never model chain
```

Comment:
This is going to let people choose gas payers that aren't on that chain accidentally, but until we can fix up our model of keys and accounts we've got to roll with this.

Reply:
It'd also be great if we could guarantee that the gas payer had gas money before kicking this off to reduce silly user errors, but not for right now.

Reply:
Is it? We pass in the chain ID

Reply:
Yup: https://github.com/kadena-io/chainweaver/blob/redesign/frontend/src/Frontend/UI/DeploymentSettings.hs#L715